src/utils.jl

```diff
@@ -0,0 +1,11 @@
+contains_path_traversal(url::URI) = contains_path_traversal(url.path)
+function contains_path_traversal(url::AbstractString)
+    # Patterns:
+    # ../, ..\, /.., \.., ./, .\, /./, \.\
+    PATH_TRAVERSAL = r"(?:^\.{2,}|\.{2,}$|\.{2,}[\/\\]|\.{1,}[\/\\]|[\/\\]\.{2,}|[\/\\]\.{1,}[\/\\])"
```

Comment:
Sorry for not noticing this earlier. Would it be better to declare it as a const outside the function scope, to avoid the cost of compiling the regex on each call?